fixes/dua-2.rb

```diff
@@ -3,21 +3,13 @@
     (contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "circa") or
      contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "ca") or
      contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "ca.") or
+     contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "c.") or
+     starts-with(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "c.") or
      contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "approximately") or
      contains(translate(./text(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "approx"))
-    and @certainty != "approximate"]').each do |ud|
-    ud['certainty'] = 'approximate'
-    a = ud.text
-    a.slice! 'circa'
-    a.slice! 'ca'
-    a.slice! 'ca.'
-    a.slice! 'approximately'
-    a.slice! 'approx'
-    a = a.tr('.', '') if a.include? "."
-    a = a.tr(',', '') if a.include? ","
-    a = a.tr('[]', '').tr('?', '')
-    a.slice! 'notebook' if a.include? "notebook"
-    a.strip!
-    ud.content = a
+      and @certainty != "approximate"]').each do |ud|
+      a = ud.text
+      a = a.gsub!(/\b(?:c(?:a)|circa|c.|ca|approx(?:imately)?|\?)[,.]?/, '').gsub(/[\[\]]/, '').strip.gsub(/\s+/, ' ')
```

Comment:
This is close, but isn't quite doing what it's supposed to: 

```ruby
/\b(?:c(?:a)/ # ...rest of regex 
```
should be
```ruby
/\b(?:c(?:a)?/ # ...rest of regex
```
and the literal '?' at the end of the alternation seems incorrect.

I think a correct regex would be:
```ruby
/\b(?:c(?:a)?|circa|c.|ca|approx(?:imately)?)[,.]?/
```

And, actually, that has some problems - I think alternation is a misleading/confusing solution here, because this will actually pull the 'c' off of circa.

```ruby
/\b(circa|ca?|approx(imately)?)\b[,.]?/
```
Might do it (any alternate that contains another must come before the one it contains, plus checking for word boundary before punc).